src/v/cluster/self_test_rpc_types.h

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright 2022 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#pragma once
+
+#include "cluster/errc.h"
+#include "model/metadata.h"
+#include "serde/serde.h"
+#include "utils/uuid.h"
+
+namespace cluster {
+
+enum class self_test_status : int8_t { running = 0, idle, cancelled };
+
+inline std::ostream& operator<<(std::ostream& o, self_test_status sts) {
+    fmt::print(o, "{}", std::underlying_type_t<self_test_status>(sts));
+    return o;
+}
+
+struct start_test_request
+  : serde::envelope<start_test_request, serde::version<0>> {
+    using rpc_adl_exempt = std::true_type;
+    uuid_t id;
+    friend std::ostream&
+    operator<<(std::ostream& o, const start_test_request& r) {
+        fmt::print(o, "{{id: {}}}", r.id);
+        return o;
+    }
+};
+
+struct stop_test_request
+  : serde::envelope<stop_test_request, serde::version<0>> {
+    using rpc_adl_exempt = std::true_type;
+    int8_t empty;
+    friend std::ostream&
+    operator<<(std::ostream& o, const stop_test_request& r) {
+        fmt::print(o, "{{empty: {}}}", r.empty);
+        return o;
+    }
+};
+
+struct test_results : serde::envelope<test_results, serde::version<0>> {
+    long disk_test_stoptime{0};
```

Comment:
Why a `long` rather than a time type?  If we don't have serialization for time types should probably add it, rather than using generic integers